[v1.0][W10-B3] HackEat

jaxony opened the pull request:
[Team Repo](https://github.com/CS2103Jan2018-W10-B3/main) 
For updates to master branch

@samzx @tohcheryl

tshradheya commented on docs/DeveloperGuide.adoc:
```diff
@@ -800,15 +803,67 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 |`* * *` |new user |see usage instructions |refer to instructions when I forget how to use the App
 
-|`* * *` |user |add a new person |
+|`* * *` |user |Use cmdline to add food that I like | I can get recommended meals that I will likely enjoy
```
`CLI` instead of `cmdline`

tshradheya commented on docs/DeveloperGuide.adoc:
```diff
@@ -800,15 +803,67 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+|`*` |user |See a history of what I’ve written as a GitHub-style diagram |I can see something cool
 |=======================================================================
 
 _{More to be added}_
```
Remove this.

tshradheya commented on docs/DeveloperGuide.adoc:
```diff
@@ -819,30 +874,73 @@ _{More to be added}_
 (For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
```
Change instances of `AddressBook` to your App name

tshradheya commented on docs/DeveloperGuide.adoc:
```diff
@@ -819,30 +874,73 @@ _{More to be added}_
++
+Use case resumes from Step 4.
 
 _{More to be added}_
```
Remove all these

tshradheya commented on src/test/java/seedu/address/logic/parser/AddCommandParserTest.java:
```diff
@@ -99,14 +99,6 @@ public void parse_compulsoryFieldMissing_failure() {
                 expectedMessage);
 
-        // missing email prefix
-        assertParseFailure(parser, NAME_DESC_BOB + PHONE_DESC_BOB + VALID_EMAIL_BOB + ADDRESS_DESC_BOB,
```
How about writing a test to check that command doesnt fail if email isnt entered?

tshradheya reviewed (commented):
Great job overall. Some points to note

To do:
- Add coveralls also for the project, will help you in knowing what tests to write
- Good Target User and User stories. Instead of having `users` everywhere, you can use adjectives to describe them for e.g. `**lazy users** so that they can order with a single command`
- Amazing job o NFR's and Use Cases
- After every version, update the PR description with summary of what everyone did.

tshradheya commented:
Good work updating the UG, AboutUs and Readme.
Also great work refactoring 🎉 

More things can be added to UG especially whats coming in V2.0[in the table of content](which you may decide to implement or not).
Please refer to the announcement/book and do the necessary for next week.
Each member of the team should have 1 merged PR of their enhancement atleast as its V1.2 release(It doesnt need to be perfect code, but something that atleast works).

If you guys, want me to review your code for quality checks tag me in the pr you make to your team repo.

Note: Every weekend please update your PR title to new version name and give a summary of what each of you have done for the week

samzx commented:
**Weekly Summary:**

Sam
- Added tests for last week's PR review (address and email fields being optional)
- Added Feature contribution in developer guide
- Created a Rating class as an extra property for Food (Minor v1.2)
- Created developer guide for `order` command with diagrams

jaxony commented:
**Weekly Summary 21 Mar**
- implemented minor enhancement: `userconfig` (unmerged)
- updated developer guide (merged)
- updated user guide (merged)

jaxony commented:
**Weekly Summary 14 Mar**
- individual: refactored AddressBook to HackEat
- team task: set up automated documentation publishing

tshradheya commented:
Great job with V1.2. Keep going with the same pace :)

This review will only be for your V1.2 DG, I will give you a review for your code near the end [ you can tag me in your PR to the team if you want me to do code review at an early stage also]

Points to Note:

1. 
![image](https://user-images.githubusercontent.com/12574756/37871030-59ab0610-3016-11e8-9097-51184c13c81c.png)

its `.PNG` instead of `.png`
I know this feature isn't already implemented, so when it is done, add more content to the DG for it.
Instead of telling what data structure is used, focus more on the architecture and how it works[the sequence]

2. 
![image](https://user-images.githubusercontent.com/12574756/37871050-aad7d8e2-3016-11e8-9758-4d382efe7f0a.png)

This code is really unnecessary, only put code snippets of what is unique to this feature. If you want to show parsing then you can show it using Sequence diagram instead of this code snippet.
Good job on dividing into parts 🎉 
Maybe add a few UML diagrams?

3. 
![image](https://user-images.githubusercontent.com/12574756/37871063-1bf0ad1a-3017-11e8-9fa7-6af08529ce5d.png)

DG should have more focus on `HOW` part and tell how the feature is implemented and its design in a top-down manner.
What kind of UML diagram is this? :P 
If you don't use standard annotation, then you need to mention the difference.


General Tip: 
- Add more diagrams
- Try to follow a similar style as `Undo-Redo section in DG`
- diagrams/examples in their description of how a feature is implemented
- The description of how it works could be better :)
- Elaborate more on Pros and Cons


Current documentation still has a lot of room for improvement (both user guide and developer guide). Please put more effort into it.

jaxony commented:
# Week ending: Wednesday 4 April
## Individual contributions
- finished logic / basic UI of major feature (chat-like interaction)
- finished tests for above
- merged all code into `master`
- updated `UserGuide.adoc`

## Team contributions
- updated `README.adoc`: changed `Ui.png` from mock to screenshot
- fixed `README.adoc`: fixed `.jar` name and team repo url, removed old instructions
- project management: closed, reviewed, approved, merged PRs, closed milestone, created `v1.4` release with `.jar`
- fixed `.jar` filename in `build.gradle`
- collation: set up code collation `bash` script and merged collation source code into master

samzx committed:
Removed logging from food selector and better handled failed selection.

samzx committed:
temporarily removed picking null values

samzx committed:
caught food selection errors in order command

samzx committed:
do not order food if no food was chosen - likely due to all foods being allergic

samzx committed:
catch npe in tests case

samzx committed:
updated test case for select fail

tohcheryl committed:
Update User Guide

tohcheryl committed:
Add test for copyfile

tohcheryl committed:
Add test for ChangePicCommand in AddressBookParser

tohcheryl committed:
Add edituser command word in addressbookparser

tohcheryl committed:
Add test for valid file

tohcheryl committed:
Merge pull request #103 from tohcheryl/fix/bug

Add edituser command word in addressbookparser

tohcheryl committed:
Merge branch 'master' into profilepanel

samzx committed:
Updated dev and user docs for order command

tohcheryl committed:
Load default profile image from the internet

tohcheryl committed:
Merge branch 'profilepanel' of https://github.com/tohcheryl/main into profilepanel

tohcheryl committed:
Use Apache FileUtils to copy file instead

samzx committed:
Changed default allergies and tags to more relevant defaults to pass tests where all foods are allergic to sample userprofile.

samzx committed:
Re-added default tag to fried

samzx committed:
Fixed missing allergy in edit system command test.

samzx committed:
Merge pull request #108 from samzx/order-docs-v1.5rc

Updated dev and user docs for order command

samzx committed:
first versions of ppp

samzx committed:
added future proposal section

samzx committed:
Revamped summary to include bits about hackeat

samzx committed:
added credits to extracts to writing.

samzx committed:
specified host location for REST API server

jaxony committed:
Developer Guide: Added UML diagram for UserProfile

tohcheryl committed:
Add test for User Profile Panel

jaxony committed:
Added DeveloperGuide writeup draft for add interactive

tohcheryl committed:
Update Project Portfolio Page for Cheryl Toh

tohcheryl committed:
Change name from John Doe to Cheryl Toh

tohcheryl committed:
Wrap text and change background colour of profile panel

jaxony committed:
Merge pull request #124 from jaxony/documentation/developer-guide

Developer Guide: Added UML diagram for UserProfile

samzx committed:
Merge branch 'master' into order-command

jaxony committed:
Fix #115: trailing whitespace after `add` now works

jaxony committed:
Refactored and fixed: infinite loop command for invalid multi value tag in add interactive

tohcheryl committed:
Change profile picture

jaxony committed:
add to javadoc

jaxony committed:
Fix test

tohcheryl committed:
Update message constraints in Name and Address class

tohcheryl committed:
Update UserGuide based on bugs identified

tohcheryl committed:
Update command example for UserConfigCommand

tohcheryl committed:
Fix checkstyle

jaxony committed:
Merge pull request #126 from jaxony/fix/add-interactive

Fix/add interactive

jaxony committed:
Merge pull request #99 from tohcheryl/profilepanel

Display basic profile

jaxony committed:
Moved command box below result display

samzx committed:
Merge pull request #109 from samzx/ppp

Updated Project Portfolio Page for Sam Xie

samzx committed:
Merge pull request #100 from samzx/order-command

Order command - Food selector now selects food with a distributed probability

jaxony committed:
change food list width

jaxony committed:
change dimensions of ResultDisplay

jaxony committed:
WIP: Added ChatPanel

samzx committed:
added collated code for sam

samzx committed:
Lean version of order command implemented. Uses email instead of Twilio API

samzx committed:
Merge branch 'collated-sam' into new-order-command

samzx committed:
added collated code in this version

samzx committed:
removed duplicate //@@author tag

samzx committed:
added unused code

samzx committed:
Fixed to send message using address

jaxony committed:
Added Jackson's PPP

samzx committed:
Refactored documentaion to reflect new utility: email (instead of dialing)

samzx committed:
updated collate for sam

tohcheryl committed:
Update messages in commands

jaxony committed:
Merge pull request #130 from jaxony/documentation/PPP

Added Jackson's PPP

samzx committed:
Further fixed documentation to remove assertions of using voice messages

tohcheryl committed:
Merge pull request #132 from tohcheryl/fix/bugs

Update messages in commands

tohcheryl committed:
Merge branch 'master' into doc/ppp

tohcheryl committed:
Update cheryltoh ppp

tohcheryl committed:
Update cheryltoh's ppp devguide section

tohcheryl committed:
Merge pull request #125 from tohcheryl/doc/ppp

Update PPP for Cheryl Toh

tohcheryl committed:
Update cheryltoh's feature contribution

tohcheryl committed:
Merge pull request #133 from tohcheryl/doc/ppp

Update cheryltoh's feature contribution

samzx committed:
Merge pull request #131 from samzx/new-order-command

New order command - email instead of dialing

samzx committed:
Added more documentation for contribution of unsued code

samzx committed:
Update ordermanager to have an optional voice call via contacting remote server.

samzx committed:
Updated facts on order manager

tohcheryl committed:
Fix program exit bug

tohcheryl committed:
Update header comments

tohcheryl committed:
Add a fullstop

samzx committed:
Merge pull request #135 from samzx/more-docs

Added more documentation for contribution of unsued code

samzx committed:
Merge branch 'master' into order-manager

samzx committed:
Fixed exception handling for order commands

samzx committed:
Merge branch 'order-manager' of https://github.com/samzx/main into order-manager

samzx committed:
updated test cases with new try blocks

samzx committed:
fixed test results to correct output

samzx committed:
Merge pull request #138 from tohcheryl/fix/bug

Fix program exit bug

samzx committed:
Merge pull request #136 from samzx/order-manager

Order manager sends HTTP to black-box server which can handle phone call

samzx committed:
Further updated documentation to reflect current state of the app

tohcheryl committed:
Update Exceptions thrown

tohcheryl committed:
Update ChangePicCommandTest to include the throwing of a CommandException object

tohcheryl committed:
Merge pull request #142 from tohcheryl/fix/exception

Update exceptions thrown for MainApp and ChangePicCommand

samzx committed:
Specified successfull execution of command as a request instead of execution

samzx committed:
Merge pull request #141 from samzx/documentation-tidy

Further updated documentation to reflect current state of the app

samzx committed:
Updated javadoc comments

samzx committed:
Merge commit '66bec81d23adda3f754b8ae0e633920585835841' into java-doc

# Conflicts:
#	collated/functional/samzx.md

samzx committed:
Edited javadoc comment on order

samzx committed:
Updated java docs and tests.

samzx committed:
Updated user guide once more

samzx committed:
Updated Developer guide and updated collations

samzx committed:
Added instructions for manual testing, resolves #146

samzx committed:
Updated name for sam xie

samzx committed:
Updated readme to include logo

samzx committed:
added newline to eof for dev guide

samzx committed:
added link to PPP

samzx committed:
fixed missing checkstyles

samzx committed:
Merge pull request #149 from samzx/java-doc

Updated Java docs and Test cases for code authored by samzx

tohcheryl committed:
Include Missing Address Prefix test for UserConfigCommandParserTest

tohcheryl committed:
Include invalid address to invalid value test in UserConfigCommandParserTest

tohcheryl committed:
Edit duplicate user message

tohcheryl committed:
Remove line of comment

tohcheryl committed:
Add collate comment

tohcheryl committed:
Update collate files

tohcheryl committed:
Remove recently ordered from toString method in UserProfile. Update defaultprofilepic

samzx committed:
Fixed bug and added tests for rating displayString

samzx committed:
Merge pull request #150 from samzx/rating-field

Fixed bug and added tests for rating displayString

tohcheryl committed:
Shifted collate comment down

tohcheryl committed:
Add more food to Sample Data Util

tohcheryl committed:
Merge pull request #151 from tohcheryl/fix/tests

Fix tests

tohcheryl committed:
Merge pull request #139 from tohcheryl/fix/codereview

Update header comments

tohcheryl committed:
Update Developer Guide

tohcheryl committed:
Update Cheryl Toh PPP

tohcheryl committed:
Update AboutUs

tohcheryl committed:
Add Sequence Diagram for EditUser

tohcheryl committed:
fix checkstyle

tohcheryl committed:
Merge pull request #152 from tohcheryl/documentation

Update Developer Guide and PPP for Cheryl Toh

tohcheryl committed:
Update test name

tohcheryl committed:
Remove RecentFoods attribute from userprofile

jaxony committed:
Chat messages works with scroll pane and automated scrolling

jaxony committed:
added styles to label

jaxony committed:
remove profilepic.png from git

jaxony committed:
make scroll pane transparent

tohcheryl committed:
Add test for changepiccommand

jaxony committed:
fix API version to 8

tohcheryl committed:
Update price test

tohcheryl committed:
Add test for UserConfigCommand

jaxony committed:
Add error labels are now styled with error style

jaxony committed:
improved UI css

jaxony committed:
remove comment

tohcheryl committed:
Update User Guide

tohcheryl committed:
Update Developer Guide to include manual testing instructions for edituser and changepic

tohcheryl committed:
Merge pull request #153 from tohcheryl/fix/removeRecentFood

Remove recent food attribute in userprofile

jaxony committed:
Message alignment works

jaxony committed:
Refactor

tohcheryl committed:
Limit length of allergy

tohcheryl committed:
Update edit command in userguide

jaxony committed:
Removing resultDisplay

tohcheryl committed:
Fix checkstyle

jaxony committed:
Added ChatPanelHandle

tohcheryl committed:
Add XmlAdaptedUserProfileTest

jaxony committed:
Fixed warning in Edit system test by using javafx thread

jaxony committed:
Replace resultDisplay with chatPanel

jaxony committed:
Fix address book system test

jaxony committed:
fix checkstyle import

jaxony committed:
Merge pull request #155 from jaxony/feature/chat-box

Chat box UI

tohcheryl committed:
Merge pull request #154 from tohcheryl/fixes

Update Documentation and Tests

jaxony committed:
rename field to classObj

tohcheryl committed:
Update profilepicture panel in MainWindow

tohcheryl committed:
Crops center of a non-square image

jaxony committed:
Implemented optional fields for Email, Address, Price, and Rating for interact add command

jaxony committed:
fix checkstyle

jaxony committed:
Merge pull request #157 from jaxony/fix/optional-add

Bug fix: Optional fields in AddCommand can now be skipped

jaxony committed:
rename food variable

jaxony committed:
Fix collation format

jaxony committed:
rename food variable

jaxony committed:
fix / remove comments inside code

jaxony committed:
Refactor to use string for field instead of class

jaxony committed:
Remove magic numbers

jaxony committed:
add collation tags

tohcheryl committed:
Make profile panel scrollable

tohcheryl committed:
Change minimum width of profilepanel

tohcheryl committed:
Add comments for methods in UserProfilePanel

jaxony committed:
Merge pull request #158 from jaxony/fix/code-review

Fix/code review

tohcheryl committed:
Remove curly braces

tohcheryl committed:
Merge pull request #159 from tohcheryl/fix/profilepanel

Fix profile panel bug

tohcheryl committed:
Merge pull request #160 from tohcheryl/test/xmluserprofile

Add test for XmlAdaptedUserProfile

tohcheryl committed:
Change png to PNG

tohcheryl committed:
Resize panels

samzx committed:
Refactored emailer out of order manager

samzx committed:
Renamed emailer to email manager

tohcheryl committed:
Modify padding and set alignment

tohcheryl committed:
Merge pull request #161 from tohcheryl/fix/profilepanel

Resize panels

jaxony committed:
Developer Guide: added Chat UI section

tohcheryl committed:
Update user guide for changepic

samzx committed:
added very basic test cases

samzx committed:
renamed test method

jaxony committed:
Added manual testing documentation for interactive add command

samzx committed:
Updated email manager docs

jaxony committed:
Updated user guide for interactive add command

jaxony committed:
Merge branch 'master' into documentation/developer-chat-ui

jaxony committed:
Fix bug #165: loop in interactively adding food when food exists already

samzx committed:
fixed 0 price and ratings field which resulted in not selecting a food.

jaxony committed:
Merge pull request #166 from jaxony/fix/repeated-food-add-interactive

Fix bug #165: loop in interactively adding food when food exists already

jaxony committed:
Merge pull request #162 from jaxony/documentation/developer-chat-ui

Developer Guide: added Chat UI section

samzx committed:
Merge pull request #164 from samzx/new-email

New email gives the restaurant a more detailed email summary of the order request.

samzx committed:
Merge pull request #167 from samzx/food-select-bug-fix

fixed 0 price and ratings field which resulted in not selecting a food.

tohcheryl committed:
Merge branch 'master' into doc/fix

tohcheryl committed:
Update Developer Guide for EditUser, ChangePic and manual testing

tohcheryl committed:
Add new line

tohcheryl committed:
Change diagram for ChangePicCommand

tohcheryl committed:
Merge pull request #170 from tohcheryl/doc/fix

Update Developer Guide (Cheryl Toh)

samzx committed:
Added code to check connection before ordering food. Notifies user to connect to internet.

samzx committed:
fixed missing white space.

samzx committed:
Merge pull request #172 from samzx/check-connection

Added code to check connection before ordering food.

tohcheryl committed:
Update UserGuide for changepic

jaxony committed:
Improved comment quality in Session.java

samzx committed:
Added new diagrams and refactored old

samzx committed:
updated java doc code standards

samzx committed:
updated docs

samzx committed:
update collated files

jaxony committed:
fix collation and javadoc in UserProfile

jaxony committed:
fix checkstyle

jaxony committed:
improved comments, moved methods around, changed space to newline in chat UI

jaxony committed:
Merge pull request #173 from jaxony/fix/quality-2

Fix/quality 2

tohcheryl committed:
Update PPP for Cheryl Toh

tohcheryl committed:
Update collated code

samzx committed:
refactored test cases with SLAP principles

samzx committed:
re-collated code

tohcheryl committed:
Merge pull request #174 from tohcheryl/doc/fix

Update PPP for Cheryl Toh

samzx committed:
Merge branch 'master' into code-quality-update

jaxony committed:
Improved documentation

jaxony committed:
Merge pull request #176 from jaxony/documentation/update-PPP

Improved documentation

samzx committed:
improved PPP for sam xie

jaxony committed:
fix link

samzx committed:
added links to collated code

samzx committed:
Merge pull request #175 from samzx/code-quality-update

Code quality update

samzx committed:
Fixed name of title to HackEat

tohcheryl committed:
Update header comments for functional code

samzx committed:
Merge pull request #177 from samzx/name-fix

Fixed name of title to HackEat

jaxony committed:
remove unused badges

jaxony committed:
update Ui.png

tohcheryl committed:
Update tests

tohcheryl committed:
Merge pull request #178 from jaxony/documentation/fix-PPP-link

fix PPP link

tohcheryl committed:
Merge pull request #180 from tohcheryl/code/fix

Fixes for comments

tohcheryl committed:
Merge pull request #179 from jaxony/documentation/readme-ui

Documentation/readme ui

tohcheryl committed:
Update User Guide

tohcheryl committed:
Update Developer Guide and padding

tohcheryl committed:
Merge pull request #182 from tohcheryl/fix/devguide

Update Developer Guide and padding

tohcheryl committed:
Merge pull request #181 from tohcheryl/fix/userguide

Update User Guide

tohcheryl committed:
Add full stop

tohcheryl committed:
Update Collated code

tohcheryl committed:
Add ellipsis to tag and allergy

tohcheryl committed:
Merge pull request #183 from tohcheryl/fix/userguide

Fix user guide and collate code

tohcheryl committed:
update devguide

tohcheryl committed:
Remove space

tohcheryl committed:
Merge pull request #184 from tohcheryl/fix/devguide

Update devguide

jaxony committed:
Re-collated

jaxony committed:
Merge pull request #185 from jaxony/collate-1.5

Re-collated

tohcheryl committed:
Update Price class

tohcheryl committed:
Update PPP

tohcheryl committed:
Final collate

tohcheryl committed:
Merge pull request #186 from tohcheryl/fixes

Update collate command for price class

samzx commented:
@jaxony can we close this pr 😛

jaxony commented:
@samzx still blocked :(
